### PR TITLE
fix: recluster maybe endless loop

### PR DESCRIPTION
**Pull request by zhyass**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

```
drop table if exists test_order all;
drop table if exists random_source all;
create table test_order (
    id bigint,
    id1 bigint,
    id2 bigint,
    id3 bigint,
    id4 bigint,
    id5 bigint,
    id6 bigint,
    id7 bigint,
    
    s1 varchar,
    s2 varchar,
    s3 varchar,
    s4 varchar,
    s5 varchar,
    s6 varchar,
    s7 varchar,
    s8 varchar,
    s9 varchar,
    s10 varchar,
    s11 varchar,
    s12 varchar,
    s13 varchar,
    
    d1 DECIMAL(20, 8),
    d2 DECIMAL(20, 8),
    d3 DECIMAL(20, 8),
    d4 DECIMAL(20, 8),
    d5 DECIMAL(20, 8),
    d6 DECIMAL(30, 8),
    d7 DECIMAL(30, 8),
    d8 DECIMAL(30, 8),
    d9 DECIMAL(30, 8),
    d10 DECIMAL(30, 8),
    
    insert_time datetime,
    insert_time1 datetime,
    insert_time2 datetime,
    insert_time3 datetime,
    
    i int
) CLUSTER BY(id,insert_time) row_per_block=10000;

create table random_source (
    id bigint,
    id1 bigint,
    id2 bigint,
    id3 bigint,
    id4 bigint,
    id5 bigint,
    id6 bigint,
    id7 bigint,
    
    s1 varchar,
    s2 varchar,
    s3 varchar,
    s4 varchar,
    s5 varchar,
    s6 varchar,
    s7 varchar,
    s8 varchar,
    s9 varchar,
    s10 varchar,
    s11 varchar,
    s12 varchar,
    s13 varchar,
    
    d1 DECIMAL(20, 8),
    d2 DECIMAL(20, 8),
    d3 DECIMAL(20, 8),
    d4 DECIMAL(20, 8),
    d5 DECIMAL(20, 8),
    d6 DECIMAL(30, 8),
    d7 DECIMAL(30, 8),
    d8 DECIMAL(30, 8),
    d9 DECIMAL(30, 8),
    d10 DECIMAL(30, 8),
    
    insert_time datetime,
    insert_time1 datetime,
    insert_time2 datetime,
    insert_time3 datetime,
    
    i int
) Engine = Random;

insert into test_order select * from random_source limit 10000000;

alter table test_order recluster final;
```

- Closes #issue

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/13332)

### CODE REVIEW

**Comment by dantengsky** [comment]
askbend:summary

**Comment by zhyass** [comment]
askbend:summary

**Comment by BohuTANG** [comment]
Add the case which in PR summary to logic test?

File: src/query/storages/fuse/src/operations/recluster.rs [hunk 1]
```diff
@@ -254,6 +254,14 @@ impl FuseTable {
...
             final_block_size
         };
+        eprintln!(
+            "block_count: {}, total_rows: {}, total_bytes: {}",
+            block_count, mutator.total_rows, mutator.total_bytes
+        );
+        eprintln!(
+            "block_num: {}, final_block_size: {}, partial_block_size: {}",
+            block_num, final_block_size, partial_block_size
+        );
```
Comment (dantengsky) [inline src/query/storages/fuse/src/operations/recluster.rs:264]: debug! or remove them?

**Review by dantengsky — APPROVED** [review]

**Pull request merged**